NPE in VariableListView

### Describe your problem!

Found on GooglePlayConsole for 2022.05.02-RC. Not sure whether already solved by #12980 or not.

```
java.lang.NullPointerException: 
  at java.util.Objects.requireNonNull (Objects.java:220)
  at cgeo.geocaching.utils.formulas.VariableMap.recalculate (VariableMap.java:300)
  at cgeo.geocaching.utils.formulas.VariableMap.remove (VariableMap.java:158)
  at cgeo.geocaching.utils.formulas.VariableList.removeVariable (VariableList.java:158)
  at cgeo.geocaching.ui.VariableListView$VariablesListAdapter.removeVarAt (VariableListView.java:380)
  at cgeo.geocaching.ui.VariableListView$VariablesListAdapter.lambda$onCreateViewHolder$5 (VariableListView.java:296)
  at cgeo.geocaching.ui.VariableListView$VariablesListAdapter.lambda$onCreateViewHolder$5$VariableListView$VariablesListAdapter (Unknown Source)
  at cgeo.geocaching.ui.-$$Lambda$VariableListView$VariablesListAdapter$FAGdGRunLlpx8rbyjIVxoohm8Zo.onClick (Unknown Source:4)
  at android.view.View.performClick (View.java:7455)
  at com.google.android.material.button.MaterialButton.performClick (MaterialButton.java:1119)
  at android.view.View.performClickInternal (View.java:7432)
  at android.view.View.access$3700 (View.java:835)
  at android.view.View$PerformClick.run (View.java:28810)
  at android.os.Handler.handleCallback (Handler.java:938)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loopOnce (Looper.java:201)
  at android.os.Looper.loop (Looper.java:288)
  at android.app.ActivityThread.main (ActivityThread.java:7870)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:548)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1003)
```

### How to reproduce?

unclear

### Actual result after these steps?

_No response_

### Expected result after these steps?

_No response_

### Reproducible

Unclear

### c:geo Version

2022.05.02-RC

### System information

_No response_

### Additional Information

_No response_

Another one to double check:

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performPauseActivityIfNeeded (ActivityThread.java:5160)
  at android.app.ActivityThread.performPauseActivity (ActivityThread.java:5111)
  at android.app.ActivityThread.handlePauseActivity (ActivityThread.java:5063)
  at android.app.servertransaction.PauseActivityItem.execute (PauseActivityItem.java:47)
  at android.app.servertransaction.ActivityTransactionItem.execute (ActivityTransactionItem.java:45)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:176)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2260)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loopOnce (Looper.java:210)
  at android.os.Looper.loop (Looper.java:299)
  at android.app.ActivityThread.main (ActivityThread.java:8166)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:556)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1037)
Caused by: java.lang.NullPointerException: 
  at java.util.Objects.requireNonNull (Objects.java:220)
  at cgeo.geocaching.utils.formulas.VariableList.getEntries (VariableList.java:237)
  at cgeo.geocaching.models.CacheVariableList.saveState (CacheVariableList.java:70)
  at cgeo.geocaching.VariablesViewPageFragment.checkUnsavedChanges (VariablesViewPageFragment.java:124)
  at cgeo.geocaching.VariablesViewPageFragment.onPause (VariablesViewPageFragment.java:158)
  at androidx.fragment.app.Fragment.performPause (Fragment.java:3168)
  at androidx.fragment.app.FragmentStateManager.pause (FragmentStateManager.java:632)
  at androidx.fragment.app.FragmentStateManager.moveToExpectedState (FragmentStateManager.java:314)
  at androidx.fragment.app.FragmentStore.moveToExpectedState (FragmentStore.java:112)
  at androidx.fragment.app.FragmentManager.moveToState (FragmentManager.java:1647)
  at androidx.fragment.app.FragmentManager.dispatchStateChange (FragmentManager.java:3128)
  at androidx.fragment.app.FragmentManager.dispatchPause (FragmentManager.java:3090)
  at androidx.fragment.app.FragmentController.dispatchPause (FragmentController.java:284)
  at androidx.fragment.app.FragmentActivity.onPause (FragmentActivity.java:390)
  at cgeo.geocaching.activity.AbstractActivity.onPause (AbstractActivity.java:118)
  at cgeo.geocaching.CacheDetailActivity.onPause (CacheDetailActivity.java:443)
  at android.app.Activity.performPause (Activity.java:8389)
  at android.app.Instrumentation.callActivityOnPause (Instrumentation.java:1531)
  at android.app.ActivityThread.performPauseActivityIfNeeded (ActivityThread.java:5150)
```

--- a/main/src/cgeo/geocaching/utils/formulas/VariableList.java
+++ b/main/src/cgeo/geocaching/utils/formulas/VariableList.java
@@ -235,6 +235,9 @@ public class VariableList {
     public List<VariableEntry> getEntries() {
         final List<VariableEntry> rows = new ArrayList<>();
         for (String v : this.variableList) {
+            if (this.variableMap.get(v) == null) {
+                continue;
+            }
             rows.add(new VariableEntry(
                     this.variablesSet.get(v) == null ? -1 : Objects.requireNonNull(this.variablesSet.get(v)),
                     v, Objects.requireNonNull(this.variableMap.get(v)).getFormulaString()));

--- a/main/src/cgeo/geocaching/utils/formulas/VariableMap.java
+++ b/main/src/cgeo/geocaching/utils/formulas/VariableMap.java
@@ -336,7 +336,10 @@ public class VariableMap {
     }
 
     private void recalculate(final String var) {
-        final VariableState state = Objects.requireNonNull(get(var));
+        final VariableState state = get(var);
+        if (state == null) {
+            return;
+        }
         state.state = State.OK;
         final List<List<String>> cyclesFound = new ArrayList<>();
         recalculate(state, var, true, new LinkedList<>(), cyclesFound);

